packages/client/machines/appMachine.ts

```diff
@@ -229,66 +232,175 @@ export function createAppMachine({
                             initial: 'waitingForEmail',
 
                             states: {
-                                waitingForEmail: {},
+                                waitingForEmail: {
+                                    on: {
+                                        REQUEST_PASSWORD_RESET: {
+                                            target: 'requestingPasswordReset',
+
+                                            actions: appModel.assign({
+                                                email: (_, event) =>
+                                                    event.email,
+                                            }),
+                                        },
+                                    },
+                                },
 
                                 requestingPasswordReset: {
```

Comment:
Praise: great work ! nice to see a such well managed authentication machine

Reply:
Aha! Long live state machines!